felix/bpf-gpl/skb.h

```diff
@@ -84,23 +94,12 @@ static CALI_BPF_INLINE long skb_iphdr_offset(void)
  */
 static CALI_BPF_INLINE void skb_refresh_hdr_ptrs(struct cali_tc_ctx *ctx)
 {
-	long offset = skb_iphdr_offset();
-	struct iphdr *ip =  ctx->data_start + offset;
-	CALI_DEBUG("IP id=%d s=%x d=%x\n",
-			bpf_ntohs(ip->id), bpf_ntohl(ip->saddr), bpf_ntohl(ip->daddr));
-	ctx->ip_header = ip;
-	ctx->nh = (void*)(ctx->ip_header+1);
+	ctx->ip_header = ctx->data_start + skb_iphdr_offset();
```

Comment:
Do we ever use `skb_refresh_hdr_ptrs()` before we make the split between V4 and V6? If not, we could also make the type in the context dependent on whether we compile for 4 or 6 and then we perhaps would not need that access function/macro?

The only other place where we assign to `ctx->ip_header` is in icmp code and that is :100:% ip version dependent.

Reply:
We do. Based on the result of `parse_packet_ip()` function we find out if the packet is v4 or v6, and that function needs to call `skb_refresh_hdr_ptrs()' before accessing the packet.